source/extensions/pageantjacker/pageantjacker.h

```diff
@@ -0,0 +1,55 @@
+/*!
+ * @file pageantjacker.h
+ * @brief Entry point and intialisation declrations for the pageantjacker extention.
+ */
+#ifndef _METERPRETER_SOURCE_EXTENSION_PAGEANTJACKER_PAGEANTJACKER_H
+#define _METERPRETER_SOURCE_EXTENSION_PAGEANTJACKER_PAGEANTJACKER_H
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+// TLVs
+// Output
+
+
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER 0
+
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER_STATUS		  MAKE_CUSTOM_TLV(TLV_META_TYPE_BOOL, TLV_TYPE_EXTENSION_PAGEANTJACKER, TLV_EXTENSIONS + 1)
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER_ERRORMESSAGE MAKE_CUSTOM_TLV(TLV_META_TYPE_STRING, TLV_TYPE_EXTENSION_PAGEANTJACKER, TLV_EXTENSIONS + 2)
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER_RETURNEDBLOB MAKE_CUSTOM_TLV(TLV_META_TYPE_RAW, TLV_TYPE_EXTENSION_PAGEANTJACKER, TLV_EXTENSIONS + 3)
+
+// Input
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER_SIZE_IN MAKE_CUSTOM_TLV(TLV_META_TYPE_UINT, TLV_TYPE_EXTENSION_PAGEANTJACKER, TLV_EXTENSIONS + 4)
+#define TLV_TYPE_EXTENSION_PAGEANTJACKER_BLOB_IN MAKE_CUSTOM_TLV(TLV_META_TYPE_RAW, TLV_TYPE_EXTENSION_PAGEANTJACKER, TLV_EXTENSIONS + 5)
+
+// Results from the pageant query function
+typedef struct __PAGEANTQUERYRESULTS {
+	BOOL result;
+	unsigned char *error_message;
+	byte *blob;
+	DWORD bloblength;
+} PAGEANTQUERYRESULTS;
+
+// Class and window name
+#define PAGEANT_NAME L"Pageant"
+
+#define PAGEANTJACKER_ERROR_SENDMESSAGE "The Pageant request was not processed."
+#define PAGEANTJACKER_ERROR_MAPVIEWOFFILE "Unable to obtain IPC memory address."
+#define PAGEANTJACKER_ERROR_CREATEFILEMAPPING "Unable to allocate memory for Pageant<-->Meterpreter IPC."
+#define PAGEANTJACKER_ERROR_ALLOC "Unable to allocate memory buffer."
+#define PAGEANTJACKER_ERROR_REQSTRINGBUILD "Unable to build Pageant request string."
+#define PAGEANTJACKER_ERROR_NOERROR "No error."
+#define PAGEANTJACKER_ERROR_NOTFOUND "Pageant not found."
+#define PAGEANTJACKER_ERROR_NOTFORWARDED "Not forwarded."

```

Comment:
Error messages shouldn't be in string format. They should instead have identifiers which are returned to MSF, and MSF can map them to strings and present them to the user. This keeps the size of the binary down and reduces the identifiable information stored in the binary.